### PR TITLE
Adding support for system package tool pkgutil on Solaris

### PR DESCRIPTION
**Pull request by aallrd**
Hello,

This PR is to add support for the pkgutil system package tool on Solaris.
This package tool is not an official one, it's linked to the [OpenCSW ](https://www.opencsw.org/about/) project that makes pre-compiled binaries for the SunOS platform easily available.
This project is however used a lot by Solaris systems administrators for the convenient and large binaries catalog they provide (and because of the poor quality of the default available tools...).

An issue I faced was with "installed" method that I copied from the BrewTool:
`exit_code = self._runner('test -n "$(brew ls --versions %s)"' % package_name, None)`
I don't have a mac to test this command but from what I understand it tries to check if the output of `brew ls --versions` is empty or if it contains the package we are trying to install.
This is the same way to check if a package is installed with `pkgutil` but I could not make it work because it seems that the underlying `os.system()` call does not expand the subshell command we want to run:

Python 2.7
```
>>> import os
>>> os.system('echo "$(echo "aa")"')
$(echo aa)
```

Bash shell
```
$ echo "$(echo "aa")"
aa
```

That's why I suspect that the BrewTool `installed()` method is always returning 0 since it's always testing a non-empty string.

### CODE REVIEW

File: conans/tools.py [hunk 1]
```diff
@@ -719,6 +722,19 @@ def installed(self, package_name):
         return exit_code == 0
 
 
+class PkgutilTool(object):
```
Comment (memsharded) [inline conans/tools.py]: Maybe ``PkgUtilTool``

File: conans/tools.py [hunk 2]
```diff
@@ -610,7 +611,7 @@ def __init__(self, runner=None, os_info=None, tool=None):
         self._is_up_to_date = False
         self._tool = tool or self._create_tool(os_info)
         self._tool._sudo_str = "sudo " if self._sudo else ""
-        self._tool._runner = runner or ConanRunner()
+        self._tool._runner = runner or ConanRunner(print_commands_to_output=True)
```
Comment (memsharded) [inline conans/tools.py]: Please do not change this default, any specific reason for this?

File: conans/tools.py [hunk 3]
```diff
@@ -10,6 +10,7 @@
 import subprocess
 import sys
 
+import io
```
Comment (memsharded) [inline conans/tools.py]: This might fail due to py2-3 differences. Maybe import from ``six``

Comment (aallrd) [inline conans/tools.py]: It seems that the io module is available for both Python 2.7 and 3.x.
Anyway, I removed it.

File: conans/tools.py (hunk 2 above)
Comment (aallrd) [inline conans/tools.py]: As I tried to explain in the PR description, I think the current way of testing for a stream ouput (not the exit code) with `test -n "$(cmd)"` for a system package tool command is not working. 
With this change I wanted to get the stdout from the command I was running to parse it and decide whether it was successful or not.

File: conans/tools.py (hunk 1 above)
Comment (aallrd) [inline conans/tools.py]: Renamed in the latest commit.

File: conans/tools.py (hunk 2 above)
Comment (memsharded) [inline conans/tools.py]: I meant, if you need to use it, it is fine, but just change the default to True in the ``PkgUtilTool`` object, not for all other tools.

**Comment by aallrd** [comment]
It works, but it will try to install a package even if it is already installed (which of course will work because it will do nothing since it's already installed :)).

It seems that for pkgutil and brew (from where I copied the test), you have to check the output of the system package tool command to understand if the package is installed or not, the return code is not enough.
Here is an example:
```
# The grep package is installed
$ pkgutil --list grep
CSWggrep
$ echo $?
0

# The toto package is not installed
$ pkgutil --list toto
$ echo $?
0

# Since the return code is useless, we test the output of the command:
$ test -n "$(pkgutil --list grep)"
$ echo $?
0

$ test -n "$(pkgutil --list toto)"
$ echo $?
1

# It seems ideal to get the correct exit code that way.
# However in Python we don't get the expected result because
# the subshell $(...) is not expanded and is considered as a string.
>>> os.system('test -n "$(pkgutil --list grep)"')
0
>>> os.system('test -n "$(pkgutil --list toto)"')
0
```

So it works in the end, but not quite as expected :/
That's why I was trying to access the output of the command to get it to work but it was a bit ugly.

**Comment by aallrd** [comment]
Actually, maybe the best solution would be to fix why the shell command we are trying to run is interpreted as what we expect.
Is there any specific reason for the usage of `os.system()` instead of a spawning a subprocess [here](https://github.com/conan-io/conan/blob/c7d66b3a083b7eec22c29f0b0f96a53f9cb96240/conans/client/runner.py#L79)?

**Comment by memsharded** [comment]
Yes, the truth is that spawning a subprocess is not a big deal, but capturing the output robustly in all OS can become problematic at some points, like process hanging (maybe the queues get full and block, don't know why) and things like that. So whenever possible, we try to use the simple os.system() approach. 

But I agree, maybe the good solution would be to do a proper subprocess with output capture.

**Comment by aallrd** [comment]
After some tests, it seems that the legacy backtick statement is expanded in the os.system() call :)

```
>>> os.system('echo "$(echo aa)"')
$(echo aa)
0
>>> os.system('echo "`echo aa`"')
aa
0
```

**Comment by memsharded** [comment]
Merged to develop (I solved one simple conflict first). It will be available next 0.26 release.
Thanks very much for your contribution!

**Pull request merged**